### ISSUE TITLE
[Panic BUG]: Client with Logger cause panic if http roundtrip return a error response!

### ISSUE THREAD

**Issue by xmac1** [elastic/go-elasticsearch#69]
file: `estransport/estransport.go:116`
```
	start := time.Now().UTC()
	res, err := c.transport.RoundTrip(req) // if err != nil, then res will be nil
	dur := time.Since(start)

	if c.logger != nil { // and only happen when client with logger specific
		var dupRes http.Response
		if res != nil {
			dupRes = *res
		}
		if c.logger.RequestBodyEnabled() {
			if req.Body != nil && req.Body != http.NoBody {
				req.Body = ioutil.NopCloser(dupReqBody)
			}
		}
		if c.logger.ResponseBodyEnabled() {
			if res.Body != nil && res.Body != http.NoBody { // if res == nil, will panic
				b1, b2, _ := duplicateBody(res.Body)
				dupRes.Body = b1
				res.Body = b2
			}
		}
		c.logger.LogRoundTrip(req, &dupRes, err, start, dur) // errcheck exclude
	}
```

**Comment by karmi** [comment]
Hey, what version of the package are you using? This should be already fixed on the current master and version branches: https://github.com/elastic/go-elasticsearch/blob/7afe9bd9f47c4fc4efe00b7e7b948f66fbc33063/estransport/estransport.go#L202

**Comment by karmi** [comment]
While investigating the issue, I've found out that the `(*esapi.Response).String()` method panics when the response is nil — it's fixed in https://github.com/elastic/go-elasticsearch/commit/ee4dede054b1bd0f214790cc46171f354a518322.

**Comment by karmi** [comment]
Hello, when you try your code with latest versions, does it still have the issue?

**Comment by orobardet** [comment]
Hello,

Got the same crash with v6 client. 
It does not seems to fixed in v6 (at least for latest tags 6.8.2).
I tried to v7 version, and its fixed.

Will there be a new version of the v6 client with this patch?
Or may I used the v7 safely to request ES 6.x?

**Comment by orobardet** [comment]
Well, it seems go getting the latest v6 tags (6.8.2) is enough to have the patch. I don't understand why, as the 6.8.2 tag is on a commit before the one including the patch on v6.x branch, but it works.

**Comment by karmi** [comment]
@orobardet , I had a look and it's most probably because https://github.com/elastic/go-elasticsearch/commit/9ca82f20e1dd5a878cf559fd295c8ea383e120f1 fixed the problem for the logger. I've released 6.8.3, so it picks up the other changes.
